test/whitened_sparse_variational.jl

```diff
@@ -0,0 +1,38 @@
+@testset "whitened_sparse_variational" begin
+    rng = MersenneTwister(123456)
+    N_cond = 5
+    N_a = 6
+    N_b = 7
+
+    # Specify prior.
+    f = GP(Matern32Kernel())
+
+    # Sample from prior.
+    x = collect(range(-1.0, 1.0; length=N_cond))
+    fx = f(x, 1e-1)
+    y = rand(rng, fx)
+
+    # Construct optimal approximate posterior.
+    fz = fx
+    q = optimal_variational_posterior(fz, fx, y)
+    Cuu = cholesky(Symmetric(cov(fz)))
+    q_ε = MvNormal(Cuu.U' \ (mean(q) - mean(fz)), Symmetric((Cuu.U' \ cov(q)) / Cuu.U))
```

Comment:
Could we not just define the non-whitened parameterisation in terms of the whitened by using this in the constructor like:

```
function SparseVariationalApproximation(fz, q)
    ...
    q_ε = MvNormal(Cuu.U' \ (mean(q) - mean(fz)), Symmetric((Cuu.U' \ cov(q)) / Cuu.U))
    WhitenedSparseVariationalApproximation(fz, q_ε)
end
```
and then just have one implementation for the posterior etc?

Reply:
Hmmm yeah, we probably could do, but I can imagine someone very occassionally actually wanting the naive parametrisation for some reason, so maybe we should keep it?

Reply:
Fair enough you might want `q` itself

Reply:
But then couldn't you just add a method to construct it if needed?

Reply:
I guess my point was more that you might want to try learning / doing inference using the unwhitened parametrisation for some reason, rather than just wanting `q(u)`.